packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -432,3 +433,39 @@ function specialCaseRefs(value: any): any {
 function undefinedIfAllValuesAreEmpty(object: object): object | undefined {
   return Object.values(object).some(v => v !== undefined) ? object : undefined;
 }
+
+export function handleAttributes(resource: CfnResource, resourceAttributes: any,
+  logicalId: string, finder: ICfnFinder): void {
+
+  const cfnParser = new CfnParser({
+    finder,
+  });
+
+  const cfnOptions = resource.cfnOptions;
+
+  cfnOptions.creationPolicy = cfnParser.parseCreationPolicy(resourceAttributes.CreationPolicy);
+  cfnOptions.updatePolicy = cfnParser.parseUpdatePolicy(resourceAttributes.UpdatePolicy);
+  cfnOptions.deletionPolicy = cfnParser.parseDeletionPolicy(resourceAttributes.DeletionPolicy);
+  cfnOptions.updateReplacePolicy = cfnParser.parseDeletionPolicy(resourceAttributes.UpdateReplacePolicy);
+  cfnOptions.metadata = cfnParser.parseValue(resourceAttributes.Metadata);
+
+  // handle DependsOn
+  resourceAttributes.DependsOn = resourceAttributes.DependsOn ?? [];
+  const dependencies: string[] = Array.isArray(resourceAttributes.DependsOn) ?
+    resourceAttributes.DependsOn : [resourceAttributes.DependsOn];
+  for (const dep of dependencies) {
+    const depResource = finder.findResource(dep);
+    if (!depResource) {
+      throw new Error(`Resource '${logicalId}' depends on '${dep}' that doesn't exist`);
+    }
+    resource.node.addDependency(depResource);
+  }
+  // handle Condition
```

Comment:
Can you actually move handling Condition above DependsOn? Yeah, I know it was the other way originally, but I think Condition first makes more sense. Thanks!